featuretools/synthesis/dfs.py

```diff
@@ -247,6 +256,33 @@ def dfs(entities=None,
     features = dfs_object.build_features(
         verbose=verbose, return_variable_types=return_variable_types)
 
+    trans_features = []
+    agg_features = []
+    groupby_features = []
+    where_features = []
+
+    base_features = get_base_features(features)
```

Comment:
I think this misses the situation where a primitive is used, but the feature that uses it is used as a stacking feature and not directly returned. In practice though, I'm not sure how frequently this edge case would show up.

I managed to force with this example, dfs warns that the `GreaterThanScalar` primitive isn't used:
```python
es = make_ecommerce_entityset()
ft.dfs(
    entityset=es,
    target_entity='régions',
    agg_primitives=['percent_true'],
    trans_primitives=[ft.primitives.GreaterThanScalar(5)],
    primitive_options={'greater_than_scalar': {'include_entities': ['stores']}},
    features_only=True
)
```
The `GreaterThanScalar` primitive is used for one of the returned features, but not as the topmost primitive:
```python
[<Feature: language>,
<Feature: PERCENT_TRUE(customers.loves_ice_cream)>,
<Feature: PERCENT_TRUE(sessions.customers.loves_ice_cream)>,
<Feature: PERCENT_TRUE(stores.num_square_feet > 5)>]
```

Reply:
Good catch. I'll look into this.

Reply:
I think I was able to fix this by changing the logic I was using to parse through the features. I added a test to make sure no warning was raised in this scenario.